packages/webview_flutter/webview_flutter_web/lib/webview_flutter_web.dart

```diff
@@ -0,0 +1,282 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:html';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:webview_flutter_platform_interface/webview_flutter_platform_interface.dart';
+import 'shims/dart_ui.dart' as ui;
+
+/// Builds an iframe based WebView.
+///
+/// This is used as the default implementation for [WebView.platform] on web.
+class WebWebViewPlatform implements WebViewPlatform {
+  /// Constructs a new instance of [WebWebViewPlatform].
+  WebWebViewPlatform() {
+    ui.platformViewRegistry.registerViewFactory(
+        'webview-iframe',
+        (int viewId) => IFrameElement()
+          ..id = 'webview-$viewId'
+          ..width = '100%'
+          ..height = '100%'
+          ..style.border = 'none');
+  }
+
+  @override
+  Widget build({
+    required BuildContext context,
+    required CreationParams creationParams,
+    required WebViewPlatformCallbacksHandler webViewPlatformCallbacksHandler,
+    required JavascriptChannelRegistry? javascriptChannelRegistry,
+    WebViewPlatformCreatedCallback? onWebViewPlatformCreated,
+    Set<Factory<OneSequenceGestureRecognizer>>? gestureRecognizers,
+  }) {
+    return HtmlElementView(
+      viewType: 'webview-iframe',
+      onPlatformViewCreated: (int viewId) {
+        if (onWebViewPlatformCreated == null) {
+          return;
+        }
+        final IFrameElement element =
+            document.getElementById('webview-$viewId')! as IFrameElement;
+        if (creationParams.initialUrl != null) {
+          element.src = creationParams.initialUrl;
+        }
+        onWebViewPlatformCreated(WebWebViewPlatformController(
+          element,
+        ));
+      },
+    );
+  }
+
+  @override
+  Future<bool> clearCookies() async => false;
+
+  /// Gets called when the plugin is registered.
+  static void registerWith(Registrar registrar) {}
```

Comment:
Shouldn't this method be registering the `instance` of the plugin to the WebWebViewPlatform class? There's no global `WebViewPlugin.instance` anymore? Or is this the new way of doing federation?

PS: let's say we want to add web support to the webview (core) example [here](https://github.com/flutter/plugins/tree/master/packages/webview_flutter/webview_flutter/example). Can it be done?

Reply:
I have to confess that I'm not entirely sure. Afaik the federation is like Stuart mentioned not very standard on this plugin. I assumed this method would not be needed, but the `generated_plugin_registrant.dart` in the example does require it to be here. @mvanbeusekom maybe you have some extra info?

Reply:
Yep, this is normally what we use to set the "instance" of the platform-specific plugin in other plugins (see [url_launcher](https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart#L46-L50) for example). However this plugin is not federated exactly the same as others, so that's why this method is empty (see [comment](https://github.com/flutter/plugins/pull/4565#issuecomment-987194281)).